nucypher/characters/control/interfaces.py

```diff
@@ -108,8 +109,9 @@ def grant(self,
                          'alice_verifying_key': new_policy.alice.stamp}
         return response_data
 
-    def revoke(self, policy_encrypting_key):
-        policy = self.character.active_policies[policy_encrypting_key]
+    def revoke(self, label: bytes, bob_verifying_key: bytes):
+        policy_id = construct_policy_id(label, bob_verifying_key)
+        policy = self.character.active_policies[policy_id]
```

Comment:
Is there a possible unhanded `KeyError` here?

Reply:
Yep, just wasn't really sure how to handle non-existent policies on the endpoint? Just return a 404? What do you think?

Reply:
Hmm - I *think* a 404 is good.  Alice not knowing about a policy is ~NotFound IMO.

Reply:
Following up <https://github.com/nucypher/nucypher/issues/904>